tests/unit-tests/UnitTests/Services/SubmitServiceTests.cs

```diff
@@ -161,15 +176,97 @@ public async Task ProcessSubmission_ShouldCallGateway_WithFormData()
             Assert.NotNull(callbackValue);
         }
 
+        [Fact]
+        public async Task PreProcessSubmission_ShouldCallGateway_CreateReferenceAndSave()
+        {
+            // Arrange
+            var schema = new FormSchemaBuilder()            
+                .WithGeneratedReference("CaseReference", "TEST")
+                .Build();
+
+            _mockSchemaFactory.Setup(_ => _.Build(It.IsAny<string>()))
+                .ReturnsAsync(schema);
+
+            _mockPageHelper
+                .Setup(_ => _.SaveCaseReference(It.IsAny<string>(), It.IsAny<string>(), true, It.IsAny<string>()))
+                .Verifiable();
+
+            // Act
+            await _service.PreProcessSubmission("form", "123454");
+
+            // Assert
+            _mockReferenceNumberProvider.Verify(_ => _.GetReference(It.IsAny<string>(), It.IsAny<int>()), Times.Once);
+            _mockPageHelper.Verify(_ => _.SaveCaseReference(It.IsAny<string>(), It.IsAny<string>(), true, "CaseReference"), Times.Once);
+        }
+
+        [Fact]
+        public async Task ProcessSubmission_ShouldReturn_GeneratedReference_IfGeneratedReferenceIsTrue()
+        {
+            // Arrange
+            var callbackValue = new ExpandoObject() as IDictionary<string, object>;
+
+            var schema = new FormSchemaBuilder()            
+                .WithGeneratedReference("CaseReference", "TEST")
+                .Build();
+
+            var questionId = "testQuestion";
+            var element = new ElementBuilder()
+                .WithQuestionId(questionId)
+                .WithType(EElementType.Textarea)
+                .Build();
+
+            var submitSlug = new SubmitSlug() { AuthToken = "AuthToken", Environment = "local", URL = "www.location.com" };
```

Comment:
() Not needed with object initialisation

Reply:
Found and fixed in 4 places!